packages/block-library/src/page-list/edit.js

```diff
@@ -23,41 +23,185 @@ import {
 // Performance of Navigation Links is not good past this value.
 const MAX_PAGE_COUNT = 100;
 const NOOP = () => {};
 
+const convertDescription = __(
+	'This menu is automatically kept in sync with pages on your site. You can manage the menu yourself by clicking customize below.'
+);
+
+function BlockContent( {
+	blockProps,
+	innerBlocksProps,
+	hasResolvedPages,
+	blockList,
+	pages,
+	parentPageID,
+} ) {
+	if ( ! hasResolvedPages ) {
+		return (
+			<div { ...blockProps }>
+				<Spinner />
+			</div>
+		);
+	}
+
+	if ( pages === null ) {
+		return (
+			<div { ...blockProps }>
+				<Notice status={ 'warning' } isDismissible={ false }>
+					{ __( 'Page List: Cannot retrieve Pages.' ) }
+				</Notice>
+			</div>
+		);
+	}
+
+	if ( pages.length === 0 ) {
+		return (
+			<div { ...blockProps }>
+				<Notice status={ 'info' } isDismissible={ false }>
+					{ __( 'Page List: Cannot retrieve Pages.' ) }
+				</Notice>
+			</div>
+		);
+	}
+
+	if ( blockList.length === 0 ) {
+		const parentPageDetails = pages.find(
+			( page ) => page.id === parentPageID
+		);
+		return (
+			<div { ...blockProps }>
+				<Warning>
+					{ sprintf(
+						// translators: %s: Page title.
+						__( '"%s" page has no children.' ),
+						parentPageDetails.title.rendered
+					) }
+				</Warning>
+			</div>
+		);
+	}
+
+	if ( pages.length > 0 ) {
+		return <ul { ...innerBlocksProps }></ul>;
+	}
+}
+
+function ConvertToLinksModal( { onClick, disabled } ) {
+	const [ isOpen, setOpen ] = useState( false );
+	const openModal = () => setOpen( true );
+	const closeModal = () => setOpen( false );
+
+	return (
+		<>
+			<BlockControls group="other">
+				<ToolbarButton title={ __( 'Edit' ) } onClick={ openModal }>
+					{ __( 'Edit' ) }
+				</ToolbarButton>
+			</BlockControls>
+			{ isOpen && (
+				<Modal
+					closeLabel={ __( 'Close' ) }
+					onRequestClose={ closeModal }
+					title={ __( 'Customize this menu' ) }
+					className={ 'wp-block-page-list-modal' }
+					aria={ {
+						describedby: 'wp-block-page-list-modal__description',
+					} }
+				>
+					<p id={ 'wp-block-page-list-modal__description' }>
+						{ convertDescription }
+					</p>
+					<div className="wp-block-page-list-modal-buttons">
+						<Button variant="tertiary" onClick={ closeModal }>
+							{ __( 'Cancel' ) }
+						</Button>
+						<Button
+							variant="primary"
+							disabled={ disabled }
+							onClick={ onClick }
+						>
+							{ __( 'Customize' ) }
+						</Button>
+					</div>
+				</Modal>
+			) }
+		</>
+	);
+}
+
 export default function PageListEdit( {
 	context,
 	clientId,
 	attributes,
 	setAttributes,
 } ) {
 	const { parentPageID } = attributes;
-	const [ pages ] = useGetPages();
-	const { pagesByParentId, totalPages, hasResolvedPages } = usePageData();
 
-	const isNavigationChild = 'showSubmenuIcon' in context;
+	const { records: pages, hasResolved: hasResolvedPages } = useEntityRecords(
+		'postType',
+		'page',
+		{
+			per_page: -1,
```

Comment:
For some reason this was set to MAX_PAGE_COUNT. I think that's wrong so I changed it to -1, to remove the limit.

Reply:
Before this change it was set to -1. The limit is there to prevent the conversion happening if there are more than 100 items, not to prevent the display of more than 100 items.

Reply:
I've updated it to MAX_PAGE_COUNT anyway as that is probably sensible.